README.md

```diff
@@ -16,84 +16,83 @@ We should be as specific and thorough as possible when defining our style, testi
-
-One of the motivations for this rule is that it is a helpful visual clue.
+YES:
 
-> Leading underscores allow you to immediately recognize the intent of such functions...
+```solidity
+Library.function()
+```
 
-We agree that a leading underscore is a useful visual clue, and this is why we oppose using them for internal library functions that can be called from other contracts. Visually, it looks wrong.
+NO:
 
 ```solidity
 Library._function()
 ```
 
-or
+If a function should never be called from another contract, it should be marked private and its name should have a leading underscore.
+
+#### Justification
+The style guide states `Underscore Prefix for Non-external Functions and Variables`. One motivation for this rule is providing a helpful visual clue:
+
+> Leading underscores allow you to immediately recognize the intent of such functions...
+
+We agree that a leading underscore is a useful visual clue, and this is why we oppose using them for internal library functions that can be called from other contracts. Visually, it looks wrong.
 
 ```solidity
 using Library for bytes
 bytes._function()
 ```
 
-Note, we cannot remedy this by insisting on the use public functions. Whether a library functions are internal or external has important implications. From the [Solidity documentation](https://docs.soliditylang.org/en/latest/contracts.html#libraries)
-
-> ... the code of internal library functions that are called from a contract and all functions called from therein will at compile time be included in the calling contract, and a regular JUMP call will be used instead of a DELEGATECALL.
+We could remedy this by insisting on the use public functions. However, developers may prefer internal functions because they are more gas efficient to call, due to how libraries are compiled in Solidity:
 
-Developers may prefer internal functions because they are more gas efficient to call.
-
-If a function should never be called from another contract, it should be marked private and its name should have a leading underscore.
+> ... the code of internal library functions that are called from a contract and all functions called from therein will at compile time be included in the calling contract, and a regular JUMP call will be used instead of a DELEGATECALL. ([source](https://docs.soliditylang.org/en/latest/contracts.html#libraries))
 
 ### C. Additions
 
-#### 1. Prefer custom errors.
+#### 1. Use custom errors.
 
-Custom errors are in some cases more gas efficient and allow passing useful information.
+Custom errors should be used wherever possible (an exception could be introducing inconsistencies to existing code bases) - they allow for passing of useful information and are in some cases more gas efficient.
 
 #### 2. Custom error names should be CapWords style.
 
 For example, `InsufficientBalance`.
 
-#### 3. Events
-
-##### A. Events names should be past tense.
+#### 3. Events names should be past tense.
```

Comment:
I think we were getting to have too many levels which made it hard to follow

Reply:
we could also instead link some sections out to different READMEs as an alternative

Reply:
imo grouping is helpful and separate readmes would require more work, but we have talked about making a doc site. Agree we are pushing a single thing to the limit

Reply:
hmm I guess it's a bit confusing that some are grouped and some aren't. Should we group all then?